freqtrade/data/entryexitanalysis.py

```diff
@@ -186,38 +207,65 @@ def prepare_results(analysed_trades, stratname,
     return res_df
 
 
-def print_results(res_df, analysis_groups, indicator_list):
+def print_results(res_df, analysis_groups, indicator_list,
+                  rejected_signals=None, to_csv=False, csv_path=None):
     if res_df.shape[0] > 0:
         if analysis_groups:
-            _do_group_table_output(res_df, analysis_groups)
+            _do_group_table_output(res_df, analysis_groups, to_csv=to_csv, csv_path=csv_path)
+
+        if rejected_signals is not None and not rejected_signals.empty:
+            _do_rejected_signals_output(rejected_signals, to_csv=to_csv, csv_path=csv_path)
 
+        # NB this can be large for big dataframes!
         if "all" in indicator_list:
-            print(res_df)
-        elif indicator_list is not None:
+            _print_table(res_df,
+                         show_index=False,
+                         name="Indicators:",
+                         to_csv=to_csv,
+                         csv_path=csv_path)
+        elif indicator_list is not None and indicator_list:
             available_inds = []
             for ind in indicator_list:
                 if ind in res_df:
                     available_inds.append(ind)
             ilist = ["pair", "enter_reason", "exit_reason"] + available_inds
-            _print_table(res_df[ilist], sortcols=['exit_reason'], show_index=False)
+            _print_table(res_df[ilist],
+                         sortcols=['exit_reason'],
+                         show_index=False,
+                         name="Indicators:",
+                         to_csv=to_csv,
+                         csv_path=csv_path)
     else:
         print("\\No trades to show")
 
 
-def _print_table(df, sortcols=None, show_index=False):
+def _print_table(df, sortcols=None, show_index=False, name=None, to_csv=False, csv_path=None):
     if (sortcols is not None):
         data = df.sort_values(sortcols)
     else:
         data = df
 
-    print(
-        tabulate(
-            data,
-            headers='keys',
-            tablefmt='psql',
-            showindex=show_index
+    if to_csv:
+        if csv_path is not None:
+            safe_name = Path(csv_path,
+                             name.lower().replace(" ", "_").replace(":", ""))
```

Comment:
the generated names are quite ... nothing-saying - and will overwrite themselfs whenever running it again.
while this CAN be intentional (save disk space) - i don't think it should be (if they overwrite - we don't need a csv output at all).

effectively, the files will be useless the moment they are written - as you'll not remember 2 days from now what the csv file was for (and there's 0 content telling you that it was for strategyX - version Y).

As such - i'd question the csv export per se - do we really think this is useful / necessary?

Reply:
The overwriting was intentional - you can run the analysis multiple times to get the output you want, and in this process could generate tens or hundreds of CSVs.

We've already had one ticket requesting CSV export, and printing 1000+ lines to stdout isn't feasible when you might have thousands of rejected signals, so I think it's a valid feature.